packages/framework-core/src/services/event-store.ts

```diff
@@ -172,4 +170,15 @@ export class EventStore {
       }
     }
   }
+
+  private async migrateSnapshot(eventEnvelope: EventEnvelope | null): Promise<EventEnvelope | null> {
+    if (eventEnvelope) {
+      const migratedEventEnvelope = await this.migrator.migrate(eventEnvelope)
+      if (eventEnvelope.version !== migratedEventEnvelope.version) {
+        await this.storeSnapshot(migratedEventEnvelope)
+      }
+      return migratedEventEnvelope
+    }
+    return null
+  }
```

Comment:
Let's avoid using `null`, it is considered bad practice. Use `undefined` instead

```suggestion
  private async migrateSnapshot(eventEnvelope?: EventEnvelope): Promise<EventEnvelope | undefined> {
    if (eventEnvelope) {
      const migratedEventEnvelope = await this.migrator.migrate(eventEnvelope)
      if (eventEnvelope.version !== migratedEventEnvelope.version) {
        await this.storeSnapshot(migratedEventEnvelope)
      }
      return migratedEventEnvelope
    }
    return undefined
  }
```